test/StudentRegistryV2.js

```diff
@@ -55,11 +68,59 @@ describe("StudentRegistryV2 Test Suite", () => {
+          ).to.be.revertedWith("Owner is excluded");
+
+          console.log("Deployed contract address:", deployedStudentRegistryV2.address);
+          console.log(deployedStudentRegistryV2);
         });
+
         it("should revert attempt to proceed without paying 1ETH as fee", async () => {
           const { deployedStudentRegistryV2, addr1 } = await loadFixture(deployUtil);
-         await expect(deployedStudentRegistryV2.connect(addr1).payFee({value: convertEther('0')})).to.be.revertedWith("You must pay fee")
+          await expect(
+            deployedStudentRegistryV2.connect(addr1).payFee({ value: convertEther("0") })
+          ).to.be.revertedWith("You must pay fee");
+        });
+      });
+
+      describe("Success", () => {
+        it.only("should successfully process payment of 1 ETH fee", async () => {
+          const { deployedStudentRegistryV2, addr1, deployedStudentRegistryV2Address, owner } = await loadFixture(
+            deployUtil
+          );
+
+          // Fetch contract balance before payFee txn
+          const initialContractBalance = await getBalance(deployedStudentRegistryV2Address);
+          console.log("Contract's initial balance:", ethers.formatEther(initialContractBalance));
+
+          // Fetch addr1  balance of  before payFee txn
+          const initialPayerBalance = await getBalance(addr1.address);
+
+          console.log("Payer's initial balance:", ethers.formatEther(initialPayerBalance));
+
+          // Perform the payment
+          const tx = await deployedStudentRegistryV2.connect(addr1).payFee({ value: ethers.parseEther("1") });
+          await tx.wait(); // Wait for the transaction to be mined
+
+          // Fetch updated balances
+          const finalPayerBalance = await getBalance(addr1.address);
+          const finalContractBalance = await getBalance(deployedStudentRegistryV2Address);
+
+          console.log("Payer's final balance:", ethers.formatEther(finalPayerBalance));
+          console.log("Contract's final balance:", ethers.formatEther(finalContractBalance));
+
+          // Assert the balance changes
+          const initialPayerBalanceNum = parseFloat(ethers.formatEther(initialPayerBalance));
+          const finalPayerBalanceNum = parseFloat(ethers.formatEther(finalPayerBalance));
+          const initialContractBalanceNum = parseFloat(ethers.formatEther(initialContractBalance));
+          const finalContractBalanceNum = parseFloat(ethers.formatEther(finalContractBalance));
+
+          // Check that the payer's balance decreased by 1 ETH
+          expect(finalPayerBalanceNum).to.be.closeTo(initialPayerBalanceNum - 1, 0.01); // Use a tolerance for floating point comparison
+
+          // Check that the contract's balance increased by 1 ETH
+          expect(finalContractBalanceNum).to.be.closeTo(initialContractBalanceNum + 1, 0.01); // Use a tolerance for floating point comparison
         });
```

Comment:
Add and emit a corresponding event in `payFee` function
Test for emitted events in `payFee` function